mlir/lib/Dialect/Linalg/Transforms/Vectorization.cpp

```diff
@@ -1939,6 +1939,129 @@ vectorizeInsertSliceOpPrecondition(tensor::InsertSliceOp sliceOp,
   return success();
 }
 
+namespace {
+enum class ConvOperationKind { Conv, Pool };
+} // namespace
+
+static bool isCastOfBlockArgument(Operation *op) {
+  return isa<CastOpInterface>(op) && op->getNumOperands() == 1 &&
+         isa<BlockArgument>(op->getOperand(0));
+}
+
+// Returns true iff it is a valid conv/pooling op.
```

Comment:
I think the comment needs to be updated. Could be something like `Returns the ConvOperationKind of the op iff .... Otherwise, returns std::nullopt.`